src/icon/internal.tsx

```diff
@@ -103,12 +103,20 @@ const InternalIcon = ({
 
   const validIcon = name && Object.prototype.hasOwnProperty.call(icons, name);
 
+  function getIconName(name: IconProps.Name) {
+    if (name === 'gen-ai') {
+      return iconSize === 'small' ? icons['gen-ai-filled'] : icons['gen-ai'];
+    } else {
+      return icons[name];
+    }
```

Comment:
This logic should have a unit test.

Also, this setup allows pulling the filled icon by its direct name `<Icon name="gen-ai-filled">`. Look at this screenshot testing page, it is there: https://d21d5uik3ws71m.cloudfront.net/components/f4463ee9af1facac5acb07a052aa4305e435fe2d/index.html#/light/icon/variant-success

Reply:
So I am thinking if we should abandon the automatic path or keep pushing it and solve these issues

Reply:
I thought it was showing there because [that page](https://github.com/cloudscape-design/components/blob/main/pages/icon/icons-list.tsx#L19 ) uses unsafe `as` casting, which as you pointed out above, overrides type checks. Same reason it shows up in the icons dropdown list.

I think we should push forward and solve. I see two paths, we can either 
1) Rely on safe/correct typings and generate both the filled and unfilled version and conditionally apply them based on the size, or 
2) Figure out a way to just generate just one icon and add the fill class for the small size.

Reply:
There is a third way. We can pull custom SVGs directly in our code. Example: https://github.com/cloudscape-design/components/blob/main/src/split-panel/icons/side-position.tsx

We can keep `gen-ai` variant in the main icons list to make listing work and pull the special `gen-ai-filled` from the code